v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go

```diff
@@ -229,3 +240,40 @@ func (report *ResourceVersionsReport) generateSupportedFrom(typeName astmodel.Ty
 
 	return supportedFrom, nil
 }
+
+// Read in any front matter present in our output file, so we preserve it when writing out the new file.
+// Returns an empty string if the file doesn't exist
+func (report *ResourceVersionsReport) readFrontMatter(outputPath string) string {
+
+	if _, err := os.Stat(outputPath); os.IsNotExist(err) {
+		return ""
+	}
+
+	klog.V(2).Infof("Reading front matter from %s", outputPath)
+	data, err := ioutil.ReadFile(outputPath)
+	if err != nil {
+		return ""
+	}
+
+	var buffer strings.Builder
+
+	// Copy until we find the end of the frontmatter
+	for index, line := range strings.Split(string(data), "\n") {
+		buffer.WriteString(line)
+		buffer.WriteString("\n")
+		if index >= 2 && strings.HasPrefix(line, "---") {
+			break
+		}
+	}
+
+	return buffer.String()
+}
+
+// defaultFrontMatter returns the default frontmatter for the report if no existing file is present
```

Comment:
If no file is present? Or if no front matter is present?

Reply:
Both. Comment updated.